lib/gears/debug.lua

```diff
@@ -45,8 +47,11 @@ local function dump_raw(data, shift, tag)
     end
 
     shift = (shift or "") .. "  "
-    for k, v in pairs(data) do
-        result = result .. "\n" .. shift .. dump_raw(v, shift, k)
+    result = result .. tostring(data)
+    if depth > 0 then
+        for k, v in pairs(data) do
+            result = result .. "\n" .. shift .. dump_raw(v, shift, k, depth - 1)
+        end
     end
```

Comment:
Good point.
Using `tostring` would not help with trying to keep it short though..